crates/nu-command/src/path/relative_to.rs

```diff
@@ -144,17 +144,84 @@ path."#
 fn relative_to(path: &Path, span: Span, args: &Arguments) -> Value {
     let lhs = expand_to_real_path(path);
     let rhs = expand_to_real_path(&args.path.item);
+
     match lhs.strip_prefix(&rhs) {
         Ok(p) => Value::string(p.to_string_lossy(), span),
-        Err(e) => Value::error(
-            ShellError::CantConvert {
-                to_type: e.to_string(),
-                from_type: "string".into(),
+        Err(e) => {
+            // On case-insensitive filesystems, try case-insensitive comparison
+            if is_case_insensitive_filesystem() {
+                if let Some(relative_path) = try_case_insensitive_strip_prefix(&lhs, &rhs) {
+                    return Value::string(relative_path.to_string_lossy(), span);
+                }
+            }
+
+            Value::error(
+                ShellError::CantConvert {
+                    to_type: e.to_string(),
+                    from_type: "string".into(),
+                    span,
+                    help: None,
+                },
                 span,
-                help: None,
-            },
-            span,
-        ),
+            )
+        }
+    }
+}
+
+/// Check if the current filesystem is typically case-insensitive
+fn is_case_insensitive_filesystem() -> bool {
```

Comment:
I'm not sure if hardcoding this based on the platform is a good idea. You can access NTFS or APFS from Linux, isn't there a better way to check the path belonging to a case insensitive fs?